lib/ObjWriter/objwriter.cpp

```diff
@@ -266,25 +266,33 @@ extern "C" bool CreateDataSection(ObjectWriter *OW, const char *SectionName,
 
   MCSection *Section = nullptr;
   SectionKind Kind =
-      IsReadOnly ? SectionKind::getReadOnly() : SectionKind::getData();
+      IsExecutable ? SectionKind::getText() :
+          IsReadOnly ? SectionKind::getReadOnly() : SectionKind::getData();
 
   switch (TheTriple.getObjectFormat()) {
   case Triple::MachO:
-    Section = OutContext.getMachOSection("__DATA", SectionName, 0, Kind);
+    Section = OutContext.getMachOSection(IsExecutable ? "__TEXT" : "__DATA", SectionName, 0, Kind);
     break;
   case Triple::COFF: {
-    unsigned Characteristics =
-        COFF::IMAGE_SCN_CNT_INITIALIZED_DATA | COFF::IMAGE_SCN_MEM_READ;
+    unsigned Characteristics = COFF::IMAGE_SCN_MEM_READ;
 
-    if (!IsReadOnly) {
-      Characteristics |= COFF::IMAGE_SCN_MEM_WRITE;
+    if (IsExecutable) {
+      Characteristics |= COFF::IMAGE_SCN_CNT_CODE | COFF::IMAGE_SCN_MEM_EXECUTE;
+    } else if (!IsReadOnly) {
+      Characteristics |= COFF::IMAGE_SCN_CNT_INITIALIZED_DATA | COFF::IMAGE_SCN_MEM_WRITE;
     }
+    else {
```

Comment:
There are a couple of places that do not meet format requirement including your previous commit.
Can you run "python utils\ccformat.py --untidy" and address the format?